lib/blockchains/EthereumBlockchain.js

```diff
@@ -336,5 +369,95 @@ class EthereumBlockchain {
+      console.log(`Sending ${amountToTransfer.toString()} ETH to ${coinAddress}`)
+
+      // Set tx with updated amount
+      txValues.amount = amountToTransfer.toString()
+      chainTransaction.actions = [await this.chain.composeAction(chainjs.Models.ChainActionType.ValueTransfer, txValues)]
+
+      await chainTransaction.setDesiredFee(ethFee)
+      await chainTransaction.prepareToBeSigned()
+      await chainTransaction.validate()
+      await chainTransaction.sign([chainjs.HelpersEthereum.toEthereumPrivateKey(hotWallet.privateKey)])
+      const tx_result = await chainTransaction.send(chainjs.Models.ConfirmType.After001)
+      console.log(`Transaction has successfully signed and sent by ${hotWallet.address} to blockchain tx hash: ${tx_result.transactionId}`)
+      return tx_result
+    } catch (err) {
+      console.error(err)
+      return { valid: false, markAs: "failed", error: err.message }
+    }
+  }
+
+  async disableWallet(withdrawalAddresses, hotWallet) {
+    console.log("Robo wallet is going to be disabled...")
+    const result = { tokensWithdrawalTx: {}, coinsWithdrawalTx: {} }
+    if (withdrawalAddresses.tokenAddress) {
+      const tokenBalance = await this.getTokenBalance(hotWallet.address)
+      if (tokenBalance.balanceInBaseUnit.isGreaterThan(new BigNumber(0))) {
+        let contractAddress = this.envs.ethereumContractAddress.toString()
+
+        // For Lockup contract we check balance of erc20 contract
+        if (this.envs.ethereumTokenType == 'token_release_schedule') {
+          contractAddress = this.envs.ethereumApprovalContractAddress.toString()
+        }
+
+        const tokensAmount = tokenBalance.balanceInBaseUnit.toString()
+        const txObject = this.withdrawTokensTx(hotWallet.address, contractAddress, withdrawalAddresses.tokenAddress, tokensAmount)
+        console.log(`Sending ${tokensAmount} tokens to ${withdrawalAddresses.tokenAddress}`);
+
+        const tx = await this.sendTransaction(txObject, hotWallet)
+        // const tx = { transactionId: 'fake token tx' }
+
+        if (typeof tx.valid == 'undefined' || tx.valid) {
+          const msg = `Successfully sent Tx id: ${tx.transactionId}`
+          console.log(msg)
+          result.tokensWithdrawalTx = { status: "success", txHash: tx.transactionId, message: msg }
+        } else {
+          const msg = "Error during withdraw tokens tx"
+          console.error(msg)
+          result.tokensWithdrawalTx = { status: "failed", txHash: null, message: msg }
+        }
+      } else {
+        result.tokensWithdrawalTx = { status: "skipped", txHash: null, message: "Tokens balance is zero, transaction skipped" }
+      }
+    }
+
+    if (withdrawalAddresses.coinAddress &&
+      await this.enoughCoinBalanceToSendTransaction(hotWallet.address) &&
+      result.tokensWithdrawalTx.status !== "failed") {
+      const coinBalance = await this.getEthBalance(hotWallet.address)
+      if (coinBalance.isGreaterThan(new BigNumber(0.001))) {
```

Comment:
Looks like we are checking for the same condition on line 438 in `enoughCoinBalanceToSendTransaction()`

Reply:
👍 Yes, it should be removed from the 438 line